pkg/assets/server.go

```diff
@@ -0,0 +1,227 @@
+		}
+
+		// get runId from path
+		_, err = w.Write(json)
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	router.PATCH("/_apis/pipelines/workflows/:runId/artifacts", func(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+		json, err := json.Marshal(ResponseMessage{
+			Message: "success",
+		})
+		if err != nil {
+			panic(err)
+		}
+
+		// get runId from path
+		_, err = w.Write(json)
+		if err != nil {
+			panic(err)
+		}
+	})
+}
+
+func downloads(router *httprouter.Router, assetPath string) {
+	router.GET("/_apis/pipelines/workflows/:runId/artifacts", func(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+		runID := params.ByName("runId")
+		dirPath := fmt.Sprintf("%s/%s", assetPath, runID)
+
+		files, err := ioutil.ReadDir(dirPath)
+		if err != nil {
+			panic(err)
+		}
+
+		var list []NamedFileContainerResourceURL
+		for _, file := range files {
+			list = append(list, NamedFileContainerResourceURL{
+				Name:                     file.Name(),
+				FileContainerResourceURL: fmt.Sprintf("http://%s/download/%s", req.Host, runID),
+			})
+		}
+
+		json, err := json.Marshal(NamedFileContainerResourceURLResponse{
+			Count: len(list),
+			Value: list,
+		})
+		if err != nil {
+			panic(err)
+		}
+
+		// get runId from path
+		_, err = w.Write(json)
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	router.GET("/download/:container", func(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+		container := params.ByName("container")
+		itemPath := req.URL.Query().Get("itemPath")
+		dirPath := fmt.Sprintf("%s/%s/%s", assetPath, container, itemPath)
+
+		var files []ContainerItem
+		err := filepath.Walk(dirPath, func(path string, info os.FileInfo, err error) error {
+			if !info.IsDir() {
+				rel, err := filepath.Rel(dirPath, path)
+				if err != nil {
+					panic(err)
+				}
+
+				files = append(files, ContainerItem{
+					Path:            fmt.Sprintf("%s/%s", itemPath, rel),
+					ItemType:        "file",
+					ContentLocation: fmt.Sprintf("http://%s/asset/%s/%s/%s", req.Host, container, itemPath, rel),
+				})
+			}
+			return nil
+		})
+		if err != nil {
+			panic(err)
+		}
+
+		json, err := json.Marshal(ContainerItemResponse{
+			Value: files,
+		})
+		if err != nil {
+			panic(err)
+		}
+
+		// get runId from path
+		_, err = w.Write(json)
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	router.GET("/asset/*path", func(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+		path := params.ByName("path")
+		dirPath := fmt.Sprintf("%s/%s", assetPath, path)
+
+		file, err := os.Open(dirPath)
+		if err != nil {
+			panic(err)
+		}
+
+		_, err = io.Copy(w, file)
+		if err != nil {
+			panic(err)
+		}
+	})
+}
+
+func ServeAssets() {
+	router := httprouter.New()
+
+	// todo: how to name this env var?
+	assetPath := os.Getenv("ASSET_PATH")
+	if assetPath == "" {
+		absPath, err := filepath.Abs("./.act")
+		if err != nil {
+			panic(err)
+		}
+
+		assetPath = absPath
+	}
```

Comment:
```suggestion
	if p == "" {
		p = os.TempDir()
	}
```

Reply:
or
`$XDG_DATA_HOME`
https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html

Reply:
We might have another problem here.
The artifacts would be stored across act executions (which is fine) but the workflowId of act is always the 1.
That means we would override and add files to that folder on consecutive executions.
Using a folder local to the project makes this at least discoverable to the user. I like your idea of using the os tempdir, but we should then scope it with a run id.